### PR TITLE
Event Editing Feature Implemented

### PR DESCRIPTION
**Pull request by fiyinfoluwaafol**
### Description
This PR primarily consists of the implementation of the feature that allows the user to be able to edit events through 2 ways :
- by clicking on a particular event and edit the information as displayed in a modal (similar to event creation modal), then submit the changes so that it reflects on the calendar immediately.
- by dragging an event, or resizing an event to change the date/time of an event, or the duration of an event respectively

No major blockers encountered during development, except having to properly handle all day events and regular events properly, there might be some edge cases that I am yet to explore but you can expect to see code implementations that account for them.

As I prepare to work on shared events, the next PR would include a general code cleanup, and also implementing some previous feedback. This is so that I can ensure everything is cleanly written and easy to follow, such that I can focus properly on the shared events implementation.

### Milestones
This works towards the user story of:
User can edit events

### Test Plan
Here is the video showing this PR's test plan (including all discussed features):
[Event Edit Feature Shown with Manual and Calendar UI](https://www.loom.com/share/fbfcfbcaaa6041959ff240fc02d7298d?sid=ae2241d4-fe83-41ec-b3a5-8ed4fd818524)

### CODE REVIEW

File: frontend/src/components/CalendarView/CalendarView.jsx [hunk 1]
```diff
@@ -122,6 +125,40 @@ function CalendarView () {
...
+      try {
+        const updatedEvent = {
+          title: info.event.title,
+          startAt: info.event.startStr,
+          endAt: info.event.endStr,
+          description: info.event.extendedProps.description,
+          location: info.event.extendedProps.location,
+          allDay: info.event.allDay
+        }
+        const backendUrlAccess = import.meta.env.VITE_BACKEND_ADDRESS;
```
Comment (juancafe4) [inline frontend/src/components/CalendarView/CalendarView.jsx:142]: should this be in the utils file?

File: frontend/src/components/CalendarView/CalendarView.jsx [hunk 2]
```diff
@@ -157,21 +160,87 @@ function CalendarView () {
         }
         throw error;
       }
+    }
 
+    const handleEventEditButton = async (info, id) => {
+      try {
```
Comment (juancafe4) [inline frontend/src/components/CalendarView/CalendarView.jsx:166]: move your API to a utils class

File: frontend/src/components/CalendarView/CalendarView.jsx [hunk 3]
```diff
@@ -157,21 +160,87 @@ function CalendarView () {
         }
         throw error;
```
Comment (juancafe4) [inline frontend/src/components/CalendarView/CalendarView.jsx:161]: is this immediately throws an error?

File: frontend/src/components/CalendarView/CalendarView.jsx [hunk 4]
```diff
@@ -122,19 +128,119 @@ function CalendarView () {
...
+      })
+      setIsModalOpen(true);
+      setIsEdit(false);
+    }
     return (
             <>
                 <Header />
                 <div id="event-src">
-                  <button onClick={() => setIsModalOpen(true)}>New Event</button>
+                  <button onClick={newEventButton}>New Event</button>
```
Comment (jameskim08) [inline frontend/src/components/CalendarView/CalendarView.jsx:233]: nit: newEventButton doesn't sound like a function name. Maybe something like onNewEventClick or handleNewEventClick

**Review by jameskim08 — COMMENTED** [review]
I assume the TC1 part of editing (updating ICS/Calendar events) will be part of a different PR?

File: frontend/src/components/CalendarView/CalendarView.jsx (hunk 4 above)
Comment (fiyinfoluwaafol) [reply frontend/src/components/CalendarView/CalendarView.jsx:233]: Yes, the TC1 part of editing events will be a part of a different PR, that's what I am working on now, after the code cleanup I mentioned earlier

Comment (fiyinfoluwaafol) [reply frontend/src/components/CalendarView/CalendarView.jsx:233]: as for the function name, I just changed it to handleNewEventClick, thanks for the suggestion

**Pull request merged**